test/conformance/filesystem_perm_test.go

```diff
@@ -43,27 +43,40 @@ func verifyPermString(resp string, expected string) error {
 	return nil
 }
 
-func TestMustFileSystemPermissions(t *testing.T) {
+func testFileSystemPermissions(paths map[string]FilePathInfo, t *testing.T) error {
 	logger := logging.GetContextLogger(t.Name())
 	clients := setup(t)
-	for key, value := range MustFilePathSpecs {
+	for key, value := range paths {
 		resp, _, err := fetchEnvInfo(clients, logger, test.EnvImageFilePathInfoPath+"?"+test.EnvImageFilePathQueryParam+"="+key, &test.Options{})
 		if err != nil {
-			t.Fatal(err)
+			return err
 		}
 
 		var f FilePathInfo
 		err = json.Unmarshal(resp, &f)
 		if err != nil {
-			t.Fatalf("Error unmarshalling response: %v", err)
+			return fmt.Errorf("Error unmarshalling response: %v", err)
 		}
 
 		if f.IsDirectory != value.IsDirectory {
-			t.Fatalf("%s isDirectory mismatch. Expect : %t Received : %t", key, value.IsDirectory, f.IsDirectory)
+			return fmt.Errorf("%s isDirectory mismatch. Expect : %t Received : %t", key, value.IsDirectory, f.IsDirectory)
 		}
 
 		if err = verifyPermString(f.PermString[1:], value.PermString); err != nil {
-			t.Fatal(err)
+			return err
 		}
 	}
+	return nil
+}
+
+func TestMustHaveFileSystemPermissions(t *testing.T) {
+	if err := testFileSystemPermissions(MustFilePathSpecs, t); err != nil {
+		t.Fatal(err)
```

Comment:
```suggestion
		t.Error(err)
```
I don't think we need fatal, which is more aggressive.